pkg/controllers/management/etcdbackup/etcdbackup.go

```diff
@@ -157,42 +196,29 @@ func (c *Controller) doClusterBackupSync(cluster *v3.Cluster) error {
 		return nil
 	}
 
-	clusterBackups, err := c.getRecuringBackupsList(cluster)
+	backups, err := c.getBackupsList(cluster)
 	if err != nil {
 		return err
 	}
+	recurringBackups := getRecurringBackupsList(backups)
 
-	// cluster has no backups, we need to kick a new one.
-	if len(clusterBackups) == 0 {
-		logrus.Debugf("[etcd-backup] cluster [%s] has no backups, creating first backup", cluster.Name)
+	// cluster has no recurring backups, we need to create initial backup
+	if len(recurringBackups) == 0 {
+		log.Debugf("[etcd-backup] cluster [%s] has no backups, creating first backup", cluster.Name)
 		newBackup, err := c.createNewBackup(cluster)
 		if err != nil {
-			return fmt.Errorf("[etcd-backup] error while creating backup for cluster [%s]: %v", cluster.Name, err)
+			return fmt.Errorf("error while creating backup for cluster [%s]: %v", cluster.Name, err)
 		}
-		logrus.Debugf("[etcd-backup] cluster [%s] new backup is created: %s", cluster.Name, newBackup.Name)
+		log.Debugf("[etcd-backup] cluster [%s] new backup is created: %s", cluster.Name, newBackup.Name)
 		return nil
 	}
 
-	// check for backups in progress
-	if cluster.Spec.RancherKubernetesEngineConfig != nil {
-		clusterTimeout := time.Duration(cluster.Spec.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig.Timeout) * time.Second
-		for _, clusterBackup := range clusterBackups {
-			backupCreationTime, err := getBackupCreatedTime(clusterBackup)
-			if err != nil {
-				logrus.Errorf("[etcd-backup] cluster [%s] backup [%s] is missing creation time: %v", cluster.Name, clusterBackup.Name, err)
-				continue
-			}
-			// cluster backup is younger than its timeout and completion is unknown
-			// therefore, it's currently running
-			if time.Since(backupCreationTime) < clusterTimeout && !rketypes.BackupConditionCompleted.IsTrue(clusterBackup) {
-				logrus.Debugf("[etcd-backup] cluster [%s] is currently creating a backup, skipping", cluster.Name)
-				return nil
-			}
-		}
+	if anyBackupsRunning(cluster, recurringBackups) || anyBackupsQueued(recurringBackups) {
```

Comment:
Swept over this before! This is the logic I was looking for. Don't create any more recurring backups if there's one already running or queued, so they don't run concurrently. Nice job.